metrics/fidelity/fidelity.py

```diff
@@ -0,0 +1,145 @@
+                }
+            ),
+        )
+
+    def _compute(
+        self,
+        predictions=None,
+        prob_y_hat=None,
+        prob_y_hat_alpha=None,
+        null_difference=None,
+        model=None,
+        tokenizer=None,
+        input_ids=None,
+        alpha=None,
+        attention_masks=None,
+        fidelity_type="sufficiency",
+        clip=True,
+        normalization=True,
+        # reduction='mean' # discuss with sam
+    ):
+
+        if normalization:
+            if (
+                (predictions is None)
+                and (prob_y_hat is None or prob_y_hat_alpha is None or null_difference is None)
+                and (model is None or input_ids is None or alpha is None)
+            ):
+                return "Please provide either predictions or model and inputs to compute predictions"
+        else:
+            if (
+                (predictions is None)
+                and (prob_y_hat is None or prob_y_hat_alpha is None)
+                and (model is None or input_ids is None or alpha is None)
+            ):
+                return "Please provide either predictions or model and inputs to compute predictions"
+
+        if prob_y_hat is None:
+            predictions, prob_y_hat = utility.compute_predictions(
+                input_ids=input_ids, model=model, attention_masks=attention_masks
+            )
+            prob_y_hat = prob_y_hat[np.arange(len(prob_y_hat)), predictions]
+
+        if prob_y_hat_alpha is None:
+            input_ids_reduced, attention_masks_reduced = utility.reduce(
+                input_ids=input_ids, rationale=alpha, tokenizer=tokenizer, fidelity_type=fidelity_type
+            )
+
+            predictions_alpha, prob_y_hat_alpha = utility.compute_predictions(
+                input_ids=input_ids_reduced, model=model, attention_masks=attention_masks_reduced
+            )
+            prob_y_hat_alpha = prob_y_hat_alpha[np.arange(len(prob_y_hat_alpha)), predictions]
+
+        # Calculating fidelity value
+        fidelity = utility.compute_fidelity(
+            prob_y_hat=prob_y_hat,
+            prob_y_hat_alpha=prob_y_hat_alpha,
+            fidelity_type=fidelity_type,
+            dataset_level=False,
+            clip=clip,
+        )
+
+        if normalization:
+            if null_difference is None:
+                null_difference = utility.compute_null_diff(
+                    input_ids=input_ids,
+                    model=model,
+                    predictions=predictions,
+                    prob_y_hat=prob_y_hat,
+                    tokenizer=tokenizer,
+                )
+
+            # Normalizing fidelity value
+            fidelity = utility.normalization(
+                fidelity=fidelity, null_difference=null_difference, fidelity_type=fidelity_type, clip=clip
+            )
+
+        # if reduction == 'mean':
+        # 	return np.mean(fidelity)
+        # elif reduction is None:
```

Comment:
this should be addressed or removed.

Reply:
I think it should be left in. It's convenient to have this method be able to example-wise fidelity scores or a mean fidelity score across a set of examples. Just add a description for the argument in the class and have the reduction default to mean (which I think is pretty standard for pytorch)